gk/main.c

```diff
@@ -562,13 +688,14 @@ gk_proc(void *arg)
 				rte_pktmbuf_free(tx_bufs[i]);
 		}
 
-		/* 
-		 * TODO Implement the command processing as follows:
-		 * Load a set of commands from its mailbox ring, and 
-		 * process each command.
-		 *
-		 * The writers of a GK mailbox: the GK-GT unit and Dynamic config.
-		 */
+		/* Load a set of commands from its mailbox ring. */
+        	num_cmd = rte_ring_sc_dequeue_burst(instance->mb->ring,
```

Comment:
Violation of the mailbox abstraction. Create the function `mb_dequeue_burst()` in the mailbox library to provide this read. If the function turns out to be just a one-line wrap around `rte_ring_sc_dequeue_burst()`, make it inline.